server/consumer.go

```diff
@@ -1027,26 +1051,41 @@ func (o *consumer) updateDeliveryInterest(localInterest bool) bool {
 }
 
 func (o *consumer) deleteNotActive() {
-	// If we have local interest simply return.
-	if o.hasLocalInterest() {
-		return
-	}
-	o.mu.RLock()
+	o.mu.Lock()
 	if o.mset == nil {
-		o.mu.RUnlock()
+		o.mu.Unlock()
 		return
 	}
+	// Push mode just look at active.
+	if o.isPushMode() {
+		// If weare active simply return.
+		if o.active {
```

Comment:
I might read wrong but active isnt yet cross account aware?

Reply:
Correct, for push we can not use same logic since its a stream.